nodejs/nodejs-cloud-run-hello-world/app.js

```diff
@@ -0,0 +1,51 @@
+const express = require('express');
+const handlebars = require('handlebars');
+const fs = require('fs').promises;
+const metadata = require('gcp-metadata');
+
+const app = express();
+app.use('/assets', express.static('assets'));
+
+// Initialize handlebars template values for reuse.
+const data = {
+  project: process.env.GOOGLE_CLOUD_PROJECT,
+  service: process.env.K_SERVICE || '???',
+  revision: process.env.K_REVISION || '???',
+};
+let template;
+
+app.get('/', async (req, res) => {
+  // Prepare the template for use if not ready.
+  if (!template) {
+    // Only check the Cloud Run metadata server if it looks like the service
```

Comment:
Re: this comment. How does this check to see if it looks like the service is deployed to Cloud Run. Doesn't it just check for the environment variable?

Reply:
Cloud Run doesn't have the project id as an environment variable. This comment though is a little confusing, specifically around the phrase "if it looks like the service...".

Reply:
The idea is we check the metadata server if we don't have the environment variable, but the environment variable is only set if the developer explicitly sets it.

I softened the comment language because neither of these are definitive about being on Cloud Run or being local, will try to clarify it.